### ISSUE TITLE
@test "checkpoint --lazy-pages and restore" fails on Ubuntu 18.04

### ISSUE THREAD

**Issue by AkihiroSuda** [opencontainers/runc#1863]
```console
$ make BUILDTAGS="seccomp apparmor selinux ambient" integration
...
ok 9 checkpoint --pre-dump and restore
not ok 10 checkpoint --lazy-pages and restore
# (from function `__runc' in file tests/integration/helpers.bash, line 64,
#  in test file tests/integration/checkpoint.bats, line 213)
#   `__runc --criu "$CRIU" restore -d --work-path ./image-dir --image-path ./image-dir --lazy-pages test_busybox_restore <&60 >&51 2>&51' failed
# runc list (status=0):
# ID          PID         STATUS      BUNDLE      CREATED     OWNER
# runc spec (status=0):
#
# runc state test_busybox (status=0):
# {
#   "ociVersion": "1.0.0",
#   "id": "test_busybox",
#   "pid": 9979,
#   "status": "running",
#   "bundle": "/tmp/busyboxtest",
#   "rootfs": "/tmp/busyboxtest/rootfs",
#   "created": "2018-08-10T12:39:53.426443783Z",
#   "owner": ""
# }
# Warn  (criu/kerndat.c:658): Can't load /run/criu.kdat
# Warn  (criu/kerndat.c:697): Can't keep kdat cache on non-tempfs
# Error (criu/uffd.c:1233): uffd: Restorer unexpectedly closed the connection
# runc --criu /usr/local/sbin/criu checkpoint --lazy-pages --page-server 0.0.0.0:27277 --status-fd /tmp/lazy-pipe-vIw76l --work-path ./work-dir --image-path .
/image-dir test_busybox (status=0):
#
# runc list (status=0):
# ID          PID         STATUS      BUNDLE      CREATED     OWNER
ok 11 runc create
...
```

```console
$ git show
commit 4056a41f58e35fd428d517de1326bc465aa6a461 (HEAD -> master, upstream/master)
Merge: beadf0ec 53fddb54
Author: Michael Crosby <crosbymichael@gmail.com>
Date:   Wed Aug 1 10:48:06 2018 -0400

    Merge pull request #1830 from crosbymichael/procs
    
    Pass GOMAXPROCS to init processes

```

```console
$ uname -a
Linux suda-ws01 4.15.0-30-generic #32-Ubuntu SMP Thu Jul 26 17:42:43 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
$ cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.1 LTS"
$ ./script/check-config.sh
info: no config specified, searching for kernel config ...
info: reading kernel config from /boot/config-4.15.0-30-generic ...

Generally Necessary:
- cgroup hierarchy: properly mounted [/sys/fs/cgroup]
- apparmor: enabled and tools installed
- CONFIG_NAMESPACES: enabled
- CONFIG_NET_NS: enabled
- CONFIG_PID_NS: enabled
- CONFIG_IPC_NS: enabled
- CONFIG_UTS_NS: enabled
- CONFIG_CGROUPS: enabled
- CONFIG_CGROUP_CPUACCT: enabled
- CONFIG_CGROUP_DEVICE: enabled
- CONFIG_CGROUP_FREEZER: enabled
- CONFIG_CGROUP_SCHED: enabled
- CONFIG_CPUSETS: enabled
- CONFIG_MEMCG: enabled
- CONFIG_KEYS: enabled
- CONFIG_MACVLAN: enabled (as module)
- CONFIG_VETH: enabled (as module)
- CONFIG_BRIDGE: enabled (as module)
- CONFIG_BRIDGE_NETFILTER: enabled (as module)
- CONFIG_NF_NAT_IPV4: enabled (as module)
- CONFIG_IP_NF_FILTER: enabled (as module)
- CONFIG_IP_NF_TARGET_MASQUERADE: enabled (as module)
- CONFIG_NETFILTER_XT_MATCH_ADDRTYPE: enabled (as module)
- CONFIG_NETFILTER_XT_MATCH_CONNTRACK: enabled (as module)
- CONFIG_NF_NAT: enabled (as module)
- CONFIG_NF_NAT_NEEDED: enabled
- CONFIG_POSIX_MQUEUE: enabled

Optional Features:
- CONFIG_USER_NS: enabled
- CONFIG_SECCOMP: enabled
- CONFIG_CGROUP_PIDS: enabled
- CONFIG_MEMCG_SWAP: enabled
- CONFIG_MEMCG_SWAP_ENABLED: missing
    (note that cgroup swap accounting is not enabled in your kernel config, you can enable it by setting boot option "swapaccount=1")
- CONFIG_BLK_CGROUP: enabled
- CONFIG_BLK_DEV_THROTTLING: enabled
- CONFIG_IOSCHED_CFQ: enabled
- CONFIG_CFQ_GROUP_IOSCHED: enabled
- CONFIG_CGROUP_PERF: enabled
- CONFIG_CGROUP_HUGETLB: enabled
- CONFIG_NET_CLS_CGROUP: enabled (as module)
- CONFIG_CGROUP_NET_PRIO: enabled
- CONFIG_CFS_BANDWIDTH: enabled
- CONFIG_FAIR_GROUP_SCHED: enabled
- CONFIG_RT_GROUP_SCHED: missing
```

@adrianreber PTAL?

**Comment by adrianreber** [comment]
Which version of CRIU are you using? On my CentOS system it works with the latest version of CRIU:

```
#️  bats -t tests/integration/checkpoint.bats 
1..4
ok 1 checkpoint and restore
ok 2 checkpoint --pre-dump and restore
ok 3 checkpoint --lazy-pages and restore
ok 4 checkpoint and restore in external network namespace
```

Let me check if it breaks with older CRIU versions.

**Comment by adrianreber** [comment]
I can see this also with CRIU 3.7. With CRIU 3.8 and 3.9, 3.10 the first check is broken but the error you see is fixed.

You basically need CRIU 3.10 plus one additional commit from https://github.com/checkpoint-restore/criu/issues/520. I also added the patch to the Fedora CRIU package : https://src.fedoraproject.org/rpms/criu/c/bef9f30caec90e01f33ccb53d2ac491a01646a2f?branch=master

So you either have to checkout criu-dev right now, or use a distribution package with the patch added.

**Comment by adrianreber** [comment]
It does not fail in travis because the lazy migration test is skipped in travis because the kernel is too old.

**Comment by AkihiroSuda** [comment]
opened #1864

**Comment by AkihiroSuda** [comment]
Closed in #1864
